sandbox/libs/authn/src/main/java/org/opensearch/authn/DefaultObjectMapper.java

```diff
@@ -43,11 +52,172 @@ public static void inject(final InjectableValues.Std injectableValues) {
+    }
+
+    public static JsonNode readTree(String string) throws IOException {
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<JsonNode>() {
+                @Override
+                public JsonNode run() throws Exception {
+                    return objectMapper.readTree(string);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
     }
 
     public static String writeValueAsString(Object value, boolean omitDefaults) throws JsonProcessingException {
-        return (omitDefaults ? defaulOmittingObjectMapper : objectMapper).writeValueAsString(value);
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<String>() {
+                @Override
+                public String run() throws Exception {
+                    return (omitDefaults ? defaulOmittingObjectMapper : objectMapper).writeValueAsString(value);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (JsonProcessingException) e.getCause();
+        }
+
+    }
+
+    public static <T> T readValue(String string, TypeReference<T> tr) throws IOException {
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
+                @Override
+                public T run() throws Exception {
+                    return objectMapper.readValue(string, tr);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+
+    }
+
+    public static <T> T readValue(String string, JavaType jt) throws IOException {
+
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
+                @Override
+                public T run() throws Exception {
+                    return objectMapper.readValue(string, jt);
+                }
+            });
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+    }
+
+    public static TypeFactory getTypeFactory() {
+        return objectMapper.getTypeFactory();
+    }
+
+    public static Set<String> getFields(Class<?> cls) {
```

Comment:
For my knowlege, when will this method be used

Reply:
I removed the unused methods of this class. They can be added back in as needed for future PRs.